### PR TITLE
Disambiguate socket API pathes on Windows with user name

### PR DESCRIPTION
**Pull request by danimo**
Addresses #3411

### CODE REVIEW

**Comment by dragotin** [comment]
:+1: but @mcastroSG please test the overlay icons on windows again with beta1 once we have it.

**Comment by danimo** [comment]
@mcastroSG actually, you can use https://rotor.int.owncloud.com/view/ownCloud-client/job/ownCloud-client-win32/6009/artifact/build/ownCloud-2.2.0.6009-nightly20160421-setup.exe which uses this build. Also test sharing, pls

**Comment by dragotin** [comment]
@danimo does that also have the socketapi changes that I just merged?

**Comment by danimo** [comment]
@dragotin no

**Comment by danimo** [comment]
@dragotin I can merge this and then do a build off master if that helps.

**Comment by dragotin** [comment]
@danimo well, I think a beta1 will do :-)

**Comment by guruz** [comment]
FYI @jturcotte

File: src/gui/socketapi.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ SocketApi::SocketApi(QObject* parent)
 
     if (Utility::isWindows()) {
         socketPath = QLatin1String("\\\\.\\pipe\\")
-        + QLatin1String("ownCloud");
+        + QLatin1String("ownCloud") + '\\' + qgetenv("USERNAME");
```
Comment (guruz) [inline src/gui/socketapi.cpp]: Any escaping needed?

File: shell_integration/windows/OCUtil/RemotePathChecker.cpp [hunk 2]
```diff
@@ -31,12 +31,24 @@
...
+    }
+}
 
 // This code is run in a thread
 void RemotePathChecker::workerThreadLoop()
 {
     auto pipename = std::wstring(L"\\\\.\\pipe\\");
-    pipename += L"ownCloud";
+    pipename += L"ownCloud\\";
+    pipename += getUserName();
```
Comment (guruz) [inline shell_integration/windows/OCUtil/RemotePathChecker.cpp]: Any escaping needed?

**Comment by danimo** [comment]
I don't think any escaping would be needed.

**Comment by dragotin** [comment]
ok, :+1:  

To branch 2.2 please.

**Comment by danimo** [comment]
Obsoleted by https://github.com/owncloud/client/pull/4738 which targets 2.2

**Pull request closed without merging**